les/clientpool.go

```diff
@@ -374,14 +443,22 @@ func (f *clientPool) balanceExhausted(id enode.ID) {
 	if c == nil || !c.priority {
 		return
 	}
+	if c.priority {
+		f.priorityConnected -= c.capacity
+	}
 	c.priority = false
 	if c.capacity != f.freeClientCap {
 		f.connectedCap += f.freeClientCap - c.capacity
 		totalConnectedGauge.Update(int64(f.connectedCap))
 		c.capacity = f.freeClientCap
 		c.peer.updateCapacity(c.capacity)
 	}
-	f.ndb.delPB(id)
+	pb := f.ndb.getOrNewPB(id)
+	pb.value = 0
+	f.ndb.setPB(id, pb)
```

Comment:
Ah, I see the modification in `nodedb. setPB` why not just delete it? If the balance is 0, then the meta doesn't have meaning anymore I think.

Reply:
It might have a meaning, it was a deliberate change to keep it if not empty. One example: if meta stores an ID of the last processed incoming payment. If the balance of the client gets exhausted but a new payment arrives around the same time (and also we have been recently restarted so we are not sure which payments are processed in the current state of the node db) then it is useful to know that the exhausted balance did not include the last payment so we can add it.

Reply:
Yes, it makes sense. 

So why not add another field for payment relative? I think we should store the last applied payment sequence number for all paid clients. `BalanceMeta`is only for the extra information?

Reply:
I think it might depend on the underlying payment system. Maybe it uses a single sequence number, maybe a block number+hash of the chain where the payments are received... the `meta` field can store any kind of reference and also additional extra info.